x-pack/plugins/security_solution/public/flyout/shared/hooks/use_fetch_field_value_pair_by_event_type.ts

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { buildEsQuery } from '@kbn/es-query';
+import type { IEsSearchRequest } from '@kbn/data-plugin/public';
+import { useQuery } from '@tanstack/react-query';
+import { createFetchData } from '../utils/fetch_data';
+import { useKibana } from '../../../common/lib/kibana';
+import { inputsSelectors } from '../../../common/store';
+import { useDeepEqualSelector } from '../../../common/hooks/use_selector';
+import { useGlobalTime } from '../../../common/containers/use_global_time';
+import type { RawResponse } from '../utils/fetch_data';
+
+const QUERY_KEY = 'FetchFieldValuePairByEventType';
+
+export enum EventKind {
+  alert = 'alert',
+  asset = 'asset',
+  enrichment = 'enrichment',
+  event = 'event',
+  metric = 'metric',
+  state = 'state',
+  pipeline_error = 'pipeline_error',
+  signal = 'signal',
+}
+
+export interface EventType {
+  eventKind: EventKind;
+  include?: boolean;
+  exclude?: boolean;
+}
+
+export interface UseFetchFieldValuePairByEventTypeParams {
+  /**
+   * Highlighted field
+   */
+  field: string;
+  /**
+   * Highlighted field value
+   */
+  values: string[];
```

Comment:
good point on the documentation, I'm now using the same doc as in the cell components.

As for using a single property, I can definitely do that. I wondering though how useful it would be: we're getting the properties from [here](https://github.com/elastic/kibana/pull/158606/files#diff-d5296dcc826179bd0eda3a3c4f2eb94c62d346d2a0b8f5821c37783bdfa871b4R133-R134), so we would be there constructing an object to then deconstruct it here?

Reply:
it would just colocate related things together I guess, if you feel like the updated docs would be enough, just go for it:)

Reply:
change implemented [here](https://github.com/elastic/kibana/pull/158606/commits/2c8f2af9d3abcef7f75a92b166f5491d7641557f), thanks for the suggestion, I like it!